common/src/main/python/dlpx/virtualization/common/_common_classes.py

```diff
@@ -293,3 +299,133 @@ def from_proto(user):
                 common_pb2.RemoteUser)
         remote_user = RemoteUser(name=user.name, reference=user.reference)
         return remote_user
+
+
+class CredentialsType(IntEnum):
+    """Plugin enum for CredentialsRequest.CredentialsType to be used for plugin operations
+    and library functions.
+
+    Plugin authors should use this instead of the corresponding protobuf generated class.
```

Comment:
Does this mean that the generated classes are never used by anyone? If so, is there a way we could stop them from being generated?

Reply:
The generated protobuf classes are used by the wrapper classes in this file, which convert those generated classes into friendlier classes for use in the plugins. The `libs.retrieve_credentials` utility converts the protobuf objects returned by the callback into instances of these friendlier classes.

I see that this particular `CredentialsType` class is not used anywhere. I don't remember anymore why it was added - it was time ago. I'll just remove it.